src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorTagHelperBinderPhase.cs

```diff
@@ -33,18 +38,27 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
             //
             // The imports come logically before the main razor file and are in the order they
             // should be processed.
-            var visitor = new DirectiveVisitor(descriptors);
+            DirectiveVisitor visitor = null;
+            if (FileKinds.IsComponent(codeDocument.GetFileKind()))
+            {
+                var currentNamespace = ComputeNamespace(codeDocument);
+                visitor = new ComponentDirectiveVisitor(codeDocument.Source.FilePath, descriptors, currentNamespace);
+            }
+            else
+            {
+                visitor = new TagHelperDirectiveVisitor(descriptors);
+            }
             var imports = codeDocument.GetImportSyntaxTrees();
             if (imports != null)
             {
                 for (var i = 0; i < imports.Count; i++)
                 {
                     var import = imports[i];
-                    visitor.Visit(import.Root);
+                    visitor.Visit(import);
                 }
             }
 
-            visitor.Visit(syntaxTree.Root);
+            visitor.Visit(syntaxTree);
 
             var tagHelperPrefix = visitor.TagHelperPrefix;
```

Comment:
I didn't touch any of the tagHelperPrefix handling at this point because it only affects components when we support components in legacy views/pages. So for a component document, this will always be null.

Reply:
As long as its always null, that satisfies my concerns. Worth a comment?

Reply:
Will add.